webapp/loggedin/modules/analysis/processingChain/processingChainState.js

```diff
@@ -21,9 +22,19 @@ const getProcessingChainOrig = _getStateProp(keys.orig, {})
 
 export const getProcessingChain = _getStateProp(keys.dirty, {})
 
+export const getAttributeUuidsOtherChains = _getStateProp(keys.attributeUuidsOtherChains, [])
+
 // ====== UPDATE
 
-export const assocProcessingChain = chain => R.pipe(R.assoc(keys.orig, chain), R.assoc(keys.dirty, chain))
+export const assocProcessingChain = (chain, attributeUuidsOtherChains = null) =>
+  R.pipe(
+    R.assoc(keys.orig, chain),
+    R.assoc(keys.dirty, chain),
+    R.unless(
+      R.always(R.isNil(attributeUuidsOtherChains)),
+      R.assoc(keys.attributeUuidsOtherChains, attributeUuidsOtherChains),
+    ),
```

Comment:
do you really need unless? what if you assoc null?

Reply:
assocProcessingChain is called even by saveDirty, maybe we can avoid that call and remove the "unless" check

Reply:
then you need two different assoc methods, one init and one assoc

Reply:
good